jwql/tests/test_pipeline_tools.py

```diff
@@ -0,0 +1,181 @@
+from jwql.utils.utils import get_config
+
+
+@pytest.mark.skipif(os.path.expanduser('~') == '/home/jenkins',
+                    reason='Requires access to central storage.')
+def test_completed_pipeline_steps():
+    """Test that the list of completed pipeline steps for a file is
+    correct
+
+    Parameters
+    ----------
+    filename : str
+        File to be checked
+    """
+
+    filename = os.path.join(get_config()['filesystem'], 'jw00312', 'jw00312002001_02102_00001_nrcb4_rateints.fits')
+    completed_steps = pipeline_tools.completed_pipeline_steps(filename)
+    true_completed = OrderedDict([('group_scale', False),
+                                  ('dq_init', True),
+                                  ('saturation', True),
+                                  ('ipc', False),
+                                  ('refpix', True),
+                                  ('superbias', True),
+                                  ('persistence', True),
+                                  ('dark_current', True),
+                                  ('linearity', True),
+                                  ('firstframe', False),
+                                  ('lastframe', False),
+                                  ('rscd', False),
+                                  ('jump', True),
+                                  ('rate', True)])
+
+    assert completed_steps == true_completed
+
+
+def test_get_pipeline_steps():
+    """Test that the proper pipeline steps are returned for an
+    instrument
+    """
+
+    nrc_req_steps = pipeline_tools.get_pipeline_steps('nircam')
+    nrc_steps = ['dq_init', 'saturation', 'superbias', 'refpix', 'linearity',
+                 'persistence', 'dark_current', 'jump', 'rate']
+    not_required = ['group_scale', 'ipc', 'firstframe', 'lastframe', 'rscd']
+    nrc_dict = OrderedDict({})
+    for step in nrc_steps:
+        nrc_dict[step] = True
+    for step in not_required:
+        nrc_dict[step] = False
+    assert nrc_req_steps == nrc_dict
+
+    nrs_req_steps = pipeline_tools.get_pipeline_steps('nirspec')
+    nrs_steps = ['group_scale', 'dq_init', 'saturation', 'superbias', 'refpix', 'linearity',
+                 'dark_current', 'jump', 'rate']
+    not_required = ['ipc', 'persistence', 'firstframe', 'lastframe', 'rscd']
+    nrs_dict = OrderedDict({})
+    for step in nrs_steps:
+        nrs_dict[step] = True
+    for step in not_required:
+        nrs_dict[step] = False
+    assert nrs_req_steps == nrs_dict
+
+    miri_req_steps = pipeline_tools.get_pipeline_steps('miri')
+    miri_steps = ['dq_init', 'saturation', 'firstframe', 'lastframe',
+                  'linearity', 'rscd', 'dark_current', 'refpix', 'jump', 'rate']
+    not_required = ['group_scale', 'ipc', 'superbias', 'persistence']
+    miri_dict = OrderedDict({})
+    for step in miri_steps:
+        miri_dict[step] = True
+    for step in not_required:
+        miri_dict[step] = False
+    assert miri_req_steps == miri_dict
```

Comment:
I see that FGS and NIRISS aren't included here - is the `get_pipeline_steps` function accurate for them?

Reply:
We should add tests for them as well. The results for both should be identical to NIRCam.

Reply:
I can take care of this